src/main/host/descriptor/tcp.c

```diff
@@ -1232,14 +1237,57 @@ static void _tcp_sendShutdownFin(TCP* tcp, const Host* host) {
     }
 }
 
+// Assumes the list is sorted e.g. with _tcp_compare_sequence().
+static PacketSelectiveAcks _tcp_selective_acks_from_list(GList* sel_acks_list) {
```

Comment:
This function doesn't handle overflowing sequence numbers, but I think this is fine since the C TCP sockets use per-packet sequence numbers rather than per-byte sequence numbers like the rust TCP sockets, and we expect not to have more than 2^32 packets per socket in a simulation.